shapely/constructive.py

```diff
@@ -769,9 +772,62 @@ def remove_repeated_points(geometry, tolerance=0.0, **kwargs):
     return lib.remove_repeated_points(geometry, tolerance, **kwargs)
 
 
+def reverse_conditioned(geometry, condition=True, **kwargs):
```

Comment:
What's the rationale of exposing this method publicly? Generally this seems something that a user can easily code themselves if they need some if/else logic around calling `reverse`?